pkg/orchestrator/orchestrator.go

```diff
@@ -80,6 +80,9 @@ type PipelineService interface {
 	// requested.
 	Stop(ctx context.Context, pipelineID string, force bool) error
 
+	// OnFailure registers a handler for a pipeline.FailureEvent.
+	OnFailure(handler pipeline.FailureHandler)
```

Comment:
Wait, do we need this? The orchestrator is not using this function, is it?

Reply:
Not really, you're right. Tbh I didn't even check how's the pipeline service declared in `runtime.go`, I _assumed_ it's an interface (Java gets me from time to time), and this was the only one. I'll remove. :+1: